src/ArgumentResolver/PackageResolver.php

```diff
@@ -16,27 +16,25 @@
 use App\Entity\Package;
 use Doctrine\Persistence\ManagerRegistry;
 use Symfony\Component\HttpFoundation\Request;
-use Symfony\Component\HttpKernel\Controller\ArgumentValueResolverInterface;
+use Symfony\Component\HttpKernel\Controller\ValueResolverInterface;
 use Symfony\Component\HttpKernel\ControllerMetadata\ArgumentMetadata;
 use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
 
-class PackageResolver implements ArgumentValueResolverInterface
+class PackageResolver implements ValueResolverInterface
 {
-    public function __construct(
-        private ManagerRegistry $doctrine,
-    ) {
-    }
-
-    public function supports(Request $request, ArgumentMetadata $argument): bool
+    public function __construct(private readonly ManagerRegistry $doctrine)
     {
-        return Package::class === $argument->getType();
     }
 
     /**
      * @return iterable<Package>
      */
     public function resolve(Request $request, ArgumentMetadata $argument): iterable
     {
+        if (Package::class !== $argument->getType()) {
+            return [];
```

Comment:
As this method is currently a generator, `return;` would be the way to make it return an empty iterable.

But my suggestion is to replace `yield $package;` with `return [$package];` as done in the core resolvers.